### PR TITLE
[FIX] mass_mailing, mass_mailing_sms : Divers fixes mostly for the reporting

### PR DESCRIPTION
**Pull request by Mariska-Archielli**
Filter default reporting by name rather then by campaign
Conditionally show the campaign group by
Resolve empty body message traceback
Mobile placeholder added
hide replied column for sms

### CODE REVIEW

File: addons/mass_mailing_sms/models/mailing_mailing.py [hunk 1]
```diff
@@ -38,6 +39,13 @@ def default_get(self, fields):
     # opt_out_link
     sms_allow_unsubscribe = fields.Boolean('Include opt-out link', default=False)
 
+    @api.constrains('body_plaintext')
+    def _check_white_spaces(self):
+        """ Check if the body message is only formed by white spaces. If so, warn the user. """
+        for mailing in self:
+            if ustr(mailing.body_plaintext).isspace():
```
Comment (tde-banana-odoo) [inline addons/mass_mailing_sms/models/mailing_mailing.py:46]: :+1:

File: addons/mass_mailing_sms/report/mailing_trace_report_views.xml [hunk 2]
```diff
@@ -1,5 +1,20 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
+    <record id="mailing_sms_trace_report_view_pivot" model="ir.ui.view">
```
Comment (tde-banana-odoo) [inline addons/mass_mailing_sms/report/mailing_trace_report_views.xml:3]: Just some guidelining: mailing_trace_report_sms_view_pivot (model - suffix (sms) - type (view) - subtype (pivot) . Same for name to update accordingly.

File: odoo/tools/mail.py [hunk 3]
```diff
@@ -304,7 +304,7 @@ def html2plaintext(html, body_id=None, encoding='utf-8'):
 
     html = ustr(html)
 
-    if not html:
+    if not html or html.isspace():
```
Comment (tde-banana-odoo) [inline odoo/tools/mail.py:307]: Could be great to move it in its own commit + add a test (in odoo/addons/base/tests/test_mail.py I think, we already test plaintext2html for example in that file).

**Review by tde-banana-odoo — COMMENTED** [review]
Validation ! Some small comments :)

**Review by odony — COMMENTED** [review]
minor nitpicking, since I was summoned by the change in odoo/tools ;-)

File: odoo/tools/mail.py (hunk 3 above)
Comment (Mariska-Archielli) [reply odoo/tools/mail.py:307]: Actually I'm sorry this condition is not useful anymore. I must have simply forgot to remove it after resolving the problem another way (with api.constrains). The SMS will now never get to this point with white spaces so the check is unnecessary.
